test/js/bun/sqlite/sqlite.test.js

```diff
@@ -587,6 +587,24 @@ it("db.query()", () => {
     }
   })(domjit);
 
+  // iterate (no args)
+  let i;
+  i = 0;
+  for (const row of db.query("SELECT * FROM test").iterate()) {
+    i === 0 && expect(JSON.stringify(row)).toBe(JSON.stringify({ id: 1, name: "Hello" }));
+    i === 1 && expect(JSON.stringify(row)).toBe(JSON.stringify({ id: 2, name: "World" }));
+    i++;
+  }
+  expect(i).toBe(2);
+
+  // iterate (args)
+  i = 0;
+  for (const row of db.query("SELECT * FROM test WHERE name = $name").iterate({ $name: "World" })) {
+    i === 0 && expect(JSON.stringify(row)).toBe(JSON.stringify({ id: 2, name: "World" }));
+    i++;
+  }
+  expect(i).toBe(1);
```

Comment:
Can you add a test that checks what happens if you call iterate, stop partly through, and then call .all()?